kyo-data/shared/src/main/scala/kyo/KyoException.scala

```diff
@@ -0,0 +1,27 @@
+package kyo
+
+import kyo.*
+import kyo.Ansi.*
+import scala.util.control.NoStackTrace
+
+class KyoException private[kyo] (
+    message: Text = null,
+    cause: Text | Throwable = null
+)(using val frame: Frame) extends NoStackTrace:
+
+    override def getCause(): Throwable =
+        cause match
+            case cause: Throwable => cause
+            case _                => null
+
+    override def getMessage(): String =
+        val detail =
+            cause match
+                case _: Throwable           => Absent
+                case cause: Text @unchecked => Maybe(cause)
+
+        val msg = frame.render(("⚠️ KyoException".red.bold :: Maybe(message).toList ::: detail.toList).map(_.show)*)
```

Comment:
it may be easier to use `.getOrElse("").show` to avoid the list?

Reply:
The size of the list determines how many lines will be rendered